cmd/crates/soroban-spec-typescript/src/project_template/src/invoke.ts

```diff
@@ -100,18 +115,30 @@ export async function invoke({ method, args = [], fee = 100, signAndSend = false
     //     }; Not yet supported`
     //   )
     // }
+
+    if (!freighterAccount) {
+      throw new Error('Not connected to Freighter')
+    }
+
+    tx = await signTx(
+      SorobanClient.assembleTransaction(tx, NETWORK_PASSPHRASE, simulated) as Tx
+    );
   }
 
-  tx = await signTx(
-    SorobanClient.assembleTransaction(tx, NETWORK_PASSPHRASE, simulated) as Tx
-  );
+  const raw = await sendTx(tx, secondsToWait);
```

Comment:
@paulbellamy in an attempt to separate the `sign` and `send` logic, lines 119-125 are now inside the `if (authsCount === 1)` block. Signing & sending, as you'd expect, works just fine. But functions that require `send` without `sign` currently fail with:

```
{
    "jsonrpc": "2.0",
    "id": 1,
    "error": {
        "code": -32603,
        "message": "received exception from stellar-core"
    }
}
```

I get this when using the version of the CLI on this branch if I add a silly `become_the_recipient` function to the crowdfund contract. Here's the XDR that gets generated for the `sendTransaction` call: `AAAAAgAAAADSu8MJFOWfcDY1dw+DLEWhCcSrtuevg0onMIUgLsCmswAAAGQAAMFmAAAABQAAAAEAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAEAAAAAAAAAGAAAAAAAAAADAAAAEgAAAAE5Md3nU7e/TCOZn0TipZhXxP3sBVYQT0rLRF9Atiz2qQAAAA8AAAAUYmVjb21lX3RoZV9yZWNpcGllbnQAAAASAAAAAAAAAADSu8MJFOWfcDY1dw+DLEWhCcSrtuevg0onMIUgLsCmswAAAAAAAAAAAAAAAA==`

I can give you more info if you need it, but I'm wondering if it's already obvious to you, based on the code above.

Reply:
I confirmed that if I add an unnecessary signature, the transaction succeeds. So it has something to do with how I'm creating this transaction envelope without a signature.

Reply:
OK, transaction envelope always needs to be signed. That's how the signer pays for the transaction.